### PR TITLE
feat(EMI-1798): add partner offer note in notification panel

### PR DESCRIPTION
**Pull request by leamotta**
This PR resolves [EMI-1798]

### Description

Add partner offer note in notification panel according to the [designs](https://www.figma.com/file/RHkhhJJwixhFPltqhH9ZkW/v1.5-Saves-Partner-Offers?node-id=3866%3A56653&mode=dev)
Simulator Screenshot - iPhone SE (3rd generation) - 2024-04-16 at 12 17 33

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.
Just a check on android since I find it impossible to run on my machine setup

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Partner offer note added to notification panel - leamotta

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[EMI-1798]: https://artsyproduct.atlassian.net/browse/EMI-1798?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Activity/components/NotificationArtworkList.tsx [hunk 1]
```diff
@@ -29,6 +31,9 @@ export const NotificationArtworkList: FC<NotificationArtworkListProps> = (props)
   const artworksConnectionData = useFragment(notificationArtworkListFragment, artworksConnection)
 
   const artworks = extractNodes(artworksConnectionData)
+  const note = partnerOffer?.note
```
Comment (gkartalis) [inline src/app/Scenes/Activity/components/NotificationArtworkList.tsx:34]: suggestion(nitpick - feel free to ignore): 
```suggestion
  const { note } = partnerOffer
```

**Review by gkartalis — APPROVED** [review]

File: src/app/Scenes/Activity/components/NotificationArtworkList.tsx [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
 import { extractNodes } from "app/utils/extractNodes"
 import { FC } from "react"
+import { ImageBackground, View } from "react-native"
 import { useFragment, graphql } from "react-relay"
 
 export interface PartnerOffer {
   endAt: string
   isAvailable: boolean
   targetHref: string
+  note: string
```
Comment (MrSltun) [inline src/app/Scenes/Activity/components/NotificationArtworkList.tsx:16]: **Suggestion:** Maybe we can make the note optional?
```suggestion
  note?: string
```

File: src/app/Scenes/Activity/components/NotificationArtworkList.tsx [hunk 3]
```diff
@@ -50,6 +55,42 @@ export const NotificationArtworkList: FC<NotificationArtworkListProps> = (props)
...
         />
       )}
+      {!!note && (
+        <View
+          style={{
+            width: "100%",
+            flexDirection: "row",
+            padding: 20,
+          }}
+        >
```
Comment (MrSltun) [inline src/app/Scenes/Activity/components/NotificationArtworkList.tsx:65]: **Suggestion:** Maybe we could use the palette mobile components? 
```suggestion
       <Flex width="100%" flexDirection="row" p={2}>
```

File: src/app/Scenes/Activity/components/NotificationArtworkList.tsx (hunk 1 above)
Comment (leamotta) [reply src/app/Scenes/Activity/components/NotificationArtworkList.tsx]: I needed to add a null check still 🤔

File: src/app/Scenes/Activity/components/NotificationArtworkList.tsx (hunk 2 above)
Comment (gkartalis) [reply src/app/Scenes/Activity/components/NotificationArtworkList.tsx:16]: with this you're gonna be okay

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Partner offer note added to notification panel - leamotta)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c3bc3bb0ada706da629dad83cce8eda9313a60b5
</p>

File: src/app/Scenes/Activity/components/NotificationArtworkList.tsx [hunk 4]
```diff
@@ -50,6 +55,29 @@ export const NotificationArtworkList: FC<NotificationArtworkListProps> = (props)
           partnerOffer={partnerOffer}
         />
       )}
+      {!!note && (
+        <Flex width="100%" flexDirection="row" p={2}>
+          <Flex width="100%" flexDirection="row" backgroundColor={color("black5")} p={1}>
```
Comment (MrSltun) [inline src/app/Scenes/Activity/components/NotificationArtworkList.tsx:60]: **Nitpick:** we can use the color names with our palette mobile components 
```suggestion
          <Flex width="100%" flexDirection="row" bg="black5" p={1}>
```

File: src/app/Scenes/Activity/components/NotificationArtworkList.tsx [hunk 5]
```diff
@@ -50,6 +55,29 @@ export const NotificationArtworkList: FC<NotificationArtworkListProps> = (props)
...
+            {!!partnerIcon && (
+              <Flex marginRight={1}>
```
Comment (MrSltun) [inline src/app/Scenes/Activity/components/NotificationArtworkList.tsx:62]:
```suggestion
              <Flex mr={1}>
```

**Comment by leamotta** [comment]
@MrSltun good observation, I updated the image source!

**Review by MrSltun — APPROVED** [review]
Looks good! 🦅

**Pull request merged**